mule-extensions-api/src/main/java/org/mule/runtime/extension/api/runtime/source/SourceCallbackContext.java

```diff
@@ -112,4 +115,16 @@
    * @return The {@link SourceCallback} that generated {@code this} context
    */
   <T, A> SourceCallback<T, A> getSourceCallback();
+
+  /**
+   * Indicates that an {@link ExtensionNotification} should be fired with the desired information when the runtime takes the
+   * source result to process it.
+   *
+   * @param action the {@link NotificationActionDefinition} to use.
+   * @param data the {@link TypedValue} data to use.
+   * @since 4.1
+   */
+  default void fire(NotificationActionDefinition<?> action, TypedValue<?> data) {
```

Comment:
how do i control if the notification is to be sync or async? I don't see it as part of the notification definition, nor is a way to control that here either

Reply:
@pablolagreca suggested hiding that for now since we may deprecate it and have listeners register in a sync or async manner.

Reply:
ok, fine, but do listeners have a way of doing that right now? If not, are we sure that gap is not going to generate performance problems in insights, analytics, etc?

Reply:
No, they can't. We should discuss this further. In any case, it could be something to tackle on a different issue, whether adding support for sync/async notifications or listeners.